src/map/script.c

```diff
@@ -9013,6 +9013,32 @@ BUILDIN(getgmlevel)
 	return true;
 }
 
+/// set the group ID of the player.
+/// setgroupid( <new group id>,{<account ID> or <Player Name>} )
+/// return 1 on success, 0 if failed.
+BUILDIN(setgroupid) {
+	TBL_PC* sd;
+	int new_group = 0;
+
+	new_group = script_getnum(st, 2);
+
+	if (script_hasdata(st, 3)) {
+		if (script_isstringtype(st, 3))
+			sd = map->nick2sd(script_getstr(st, 3));
+		else
+			sd = map->id2sd(script_getnum(st, 3));
+	}
+	else
+		sd = script->rid2sd(st);
+
+	if (sd == NULL)
+		return false; // no player attached, report source
```

Comment:
like I said before, `return false;` just print the source location of the npc name
> since the script->rid2sd already has script->reportsrc(st); function

`return true;` means end of the function without displaying an error
`return false;` means end of the function + display the location and name of the source npc